### PR TITLE
entgql: generate custom scalar

### PR DESCRIPTION
**Pull request by giautm**
Related: https://github.com/ent/ent/issues/2351

### CODE REVIEW

File: entgql/schema.go [hunk 1]
```diff
@@ -78,26 +79,72 @@ var (
...
+	return &ast.Directive{
+		Name: "goModel",
+		Arguments: ast.ArgumentList{
+			{
+				Name:  "model",
+				Value: &ast.Value{Kind: ast.StringValue, Raw: model},
+			},
+		},
+	}
+}
```
Comment (giautm) [inline entgql/schema.go:120]: This may conflict with PR: https://github.com/ent/contrib/pull/249

File: entgql/extension.go [hunk 2]
```diff
@@ -42,6 +42,14 @@ import (
 )
 
 type (
+	// CustomScalar holds information about a custom scalar type.
+	CustomScalar struct {
+		Name           string
+		Description    string
+		SpecifiedByURL string
+		GoModel        string
```
Comment (giautm) [inline entgql/extension.go:50]: Ident or GoModel?

File: entgql/extension.go [hunk 3]
```diff
@@ -201,6 +209,14 @@ func WithSchemaGenerator() ExtensionOption {
 	}
 }
 
+// WithCustomScalars adds custom scalar types to the schema.
+func WithCustomScalars(scalars ...CustomScalar) ExtensionOption {
+	return func(e *Extension) error {
+		e.scalars = scalars
```
Comment (giautm) [inline entgql/extension.go:215]: Todo: add validate

File: entgql/extension.go [hunk 4]
```diff
@@ -201,6 +209,14 @@ func WithSchemaGenerator() ExtensionOption {
 	}
 }
 
+// WithCustomScalars adds custom scalar types to the schema.
+func WithCustomScalars(scalars ...CustomScalar) ExtensionOption {
```
Comment (a8m) [inline entgql/extension.go:213]: Why do you need this? This information can be extracted from the GoType field, by checking that it implements the GQL marshaller interfaces.

Comment (giautm) [reply entgql/extension.go:213]: Are all types implements the GQL marshaller interfaces is a scalar? How we can check for UUID case? It's not implement GQL marshaller interfaces, but using custom marshal functions
```
  UUID:
    model:
      - entgo.io/contrib/entgql/internal/todouuid/ent/schema/uuidgql.UUID
```
 This allow user defined their custom scalar type with more information

File: entgql/extension.go (hunk 3 above)
Comment (a8m) [reply entgql/extension.go:215]: Got it, agree.

File: entgql/extension.go (hunk 4 above)
Comment (a8m) [reply entgql/extension.go:213]: Got it, agree.

**Comment by giautm** [comment]
Let this part be handled later

**Pull request closed without merging**